Objects/frameobject.c

```diff
@@ -1004,7 +1005,11 @@ PyFrame_FastToLocalsWithError(PyFrameObject *f)
         PyErr_BadInternalCall();
         return -1;
     }
-    return _PyFrame_FastToLocalsWithError(f->f_frame);
+    int err = _PyFrame_FastToLocalsWithError(f->f_frame);
+    if (err == 0) {
+        f->f_fast_as_locals = 1;
```

Comment:
Would it possible to modify _PyFrame_FastToLocalsWithError() to set the member? Pass the PyFrameObject* to this function.

Reply:
No, it no needs to take a `_PyInterpreterFrame`: https://github.com/python/cpython/blob/main/Python/ceval.c#L3890`

Reply:
Ok. Well, at least, _PyFrame_FastToLocalsWithError() is not exported, and so cannot be used by 3rd party C extension.